src/server/server_family.h

```diff
@@ -195,8 +195,8 @@ class ServerFamily {
 
   // if new_version is true, saves DF specific, non redis compatible snapshot.
   // if basename is not empty it will override dbfilename flag.
-  GenericError DoSave(bool new_version, std::string_view basename, Transaction* transaction,
-                      bool ignore_state = false);
+  GenericError DoSave(bool new_version, std::string_view sub_path, std::string_view basename,
```

Comment:
Since the arguments of this function seem to grow, would it be worth to group them under a struct to clean up the API ?

Reply:
Also maybe even worth avoiding the split in `GetVersionSubPathAndBasename`. you can do that later when needed as long as you parse it and you know the path does not have an error. That avoid all the changes here including the tupple below :)

Reply:
Make sense, i'll do change interface to accept `GetVersionSubPathAndBasename` struct.